tasks/simplification.py

```diff
@@ -0,0 +1,156 @@
+
+    def __init__(self, *args, **kwargs):
+        super(SimplifyGeometriesMapshaper, self).__init__(*args, **kwargs)
+
+        self.table_out = '{tablename}{suffix}'.format(tablename=self.table_input, suffix=SIMPL_SUFFIX)
+        if self.table_output:
+            self.table_out = self.table_output
+
+    def requires(self):
+        return SimplifyShapefile(schema=self.schema, table_input=self.table_input, table_output=self.table_out,
+                                 retainpercentage=self.retainpercentage, skipfailures=self.skipfailures,
+                                 maxmemory=self.maxmemory)
+
+    def run(self):
+        cmd = 'PG_USE_COPY=yes ' \
+              'ogr2ogr -f PostgreSQL "PG:dbname=$PGDATABASE active_schema={schema}" ' \
+              '-t_srs "EPSG:4326" -nlt MultiPolygon -nln {table} ' \
+              '-lco OVERWRITE=yes -lco PRECISION=no -lco GEOMETRY_NAME={geomfield} ' \
+              '-lco SCHEMA={schema} {shp_path} '.format(
+                    schema=self.output().schema,
+                    table=self.output().tablename,
+                    geomfield=self.geomfield,
+                    shp_path=os.path.join(self.input().path, shp_filename(self.table_out)))
+        shell(cmd)
+
+    def output(self):
+        return PostgresTarget(self.schema, self.table_out)
+
+
+def postgis_simplification_factor(schema, table, geomfield, divisor_power):
+    session = CurrentSession().get()
+    return session.execute('SELECT AVG(ST_Perimeter({geomfield}) / ST_NPoints({geomfield})) / 10 ^ ({divisor} / 10) '
+                           'from "{schema}".{table}'.format(
+                            schema=schema, table=table, geomfield=geomfield, divisor=divisor_power
+                           )).fetchone()[0]
+
+
+class SimplifyGeometriesPostGIS(Task):
+    schema = Parameter()
+    table_input = Parameter()
+    table_output = Parameter(default='')
+    geomfield = Parameter(default=DEFAULT_GEOMFIELD)
+    retainfactor = Parameter(default=DEFAULT_P_RETAIN_FACTOR)
+
+    def __init__(self, *args, **kwargs):
+        super(SimplifyGeometriesPostGIS, self).__init__(*args, **kwargs)
+
+        self.table_out = '{tablename}{suffix}'.format(tablename=self.table_input, suffix=SIMPL_SUFFIX)
+        if self.table_output:
+            self.table_out = self.table_output
+
+    def run(self):
+        session = CurrentSession().get()
+
+        columns = session.execute("SELECT column_name "
+                                  "FROM information_schema.columns "
+                                  "WHERE table_schema = '{schema}' "
+                                  "AND table_name   = '{table}'".format(
+                                    schema=self.schema, table=self.table_input.lower())).fetchall()
+
+        factor = postgis_simplification_factor(self.schema, self.table_input, self.geomfield, self.retainfactor)
+
+        simplified_geomfield = 'ST_MakeValid(ST_SimplifyVW({geomfield}, {factor})) {geomfield}'.format(
```

Comment:
One question, we should not make here a ST_CollectionExtract(...,3) in order to get only polygon/multipolygon instead of geometrycollection which we don't support?

Reply:
The idea of these tasks is simplifying the data of an existing table (an existing `TempTableTask`) so I wouldn't expect unsupported geometry types, but as we are using the `-nlt MULTIPOLYGON` option for the `ogr2ogr` import, we could add `ST_CollectionExtract(...,3)` as you suggest for symmetry.

Reply:
My question was more in the line that `ST_Makevalid` could create `GeometryCollection` from the validated geometries as it says:

```In case of full or partial dimensional collapses, the output geometry may be a collection of lower-to-equal dimension geometries or a geometry of lower dimension.```

I've suffered from that when I have to make geometries valid. I don't know if that affect us here but just in case I prefer to ask hehehe :)

Reply:
Oh... yes! You're right! I'm updating it :)